Include/cpython/abstract.h

```diff
@@ -77,8 +80,8 @@ PyVectorcall_Function(PyObject *callable)
     assert(PyCallable_Check(callable));
     offset = tp->tp_vectorcall_offset;
     assert(offset > 0);
-    ptr = (vectorcallfunc *)(((char *)callable) + offset);
-    return *ptr;
+    vc.data = (char *)callable + offset;
+    return *vc.ptr;
```

Comment:
I don't think this is better. The correct, idiomatic solution is something like:
```c
    vectorcallfunc *ptr;
    memcpy(ptr, (char *) callable + offset, sizeof(ptr));
    return *ptr;
```

Reply:
(also, note that any decent compiler will optimize the memcpy call away)